packages/next/client/link.tsx

```diff
@@ -408,6 +408,14 @@ const Link = React.forwardRef<HTMLAnchorElement, LinkPropsReal>(
       } else {
         child = React.Children.only(children)
       }
+    } else {
+      if (process.env.NODE_ENV === 'development') {
+        if ((children as any)?.type === 'a') {
+          throw new Error(
+            'Invalid <Link> with <a> child. Please remove <a> or use <Link legacyBehavior>.\nLearn more: https://nextjs.org/docs/messages/new-link-a-tag'
```

Comment:
Let's rename this to invalid-new-link-with-extra-anchor

Reply:
Done [5d70210](https://github.com/vercel/next.js/pull/41756/commits/5d70210e9aafa05a84632241a6fd2c92d4a0fda1)